src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -416,28 +415,135 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+        # Error is due to the repository not being found
+        else:
+            raise ResourceNotFoundError(REPOSITORY_INPUT_ERROR_MESSAGE)
+
+# Validate if specific blob can be pulled
+def _check_blob_exists(cmd,
+                       registry_name,
+                       repository_name,
+                       image):
+    
+    import requests
+    from .repository import get_access_credentials
+
+    registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    login_server = registry.login_server.rstrip('/')
+
+    # Get the access credentials for the registry
+    registry_name, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+    
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+    
+    response = requests.get(
+        manifest_url,
+        headers={'Accept': 'application/vnd.oci.artifact.manifest.v1+json'
+            ', application/vnd.cncf.oras.artifact.manifest.v1+json'
+            ', application/vnd.oci.image.manifest.v1+json'
+            ', application/vnd.oci.image.index.v1+json'
+            ', application/vnd.docker.distribution.manifest.v2+json'
+            ', application/vnd.docker.distribution.manifest.list.v2+json'},
+        auth=(username, password))
+    
+    manifest = response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    def get_smallest_blob_digest(manifest):
+       
+        smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
+        smallest_blob_digest = smallest_blob['digest']
+
+        return smallest_blob_digest
+
+    digest = get_smallest_blob_digest(manifest)
+
+    # Pull blob - GET {url}/v2/{name}/blobs/{digest}
+    request_url = 'https://' + login_server + '/v2/' + repository_name + '/blobs/' + digest
+    
+    logger.debug(add_timestamp("Sending a HTTP GET request to {}".format(request_url)))
+    
+    response = requests.get(request_url, auth=(username, password))
+    
+    if response.status_code < 400:
+        print_pass("Blob identified by image '{}' is available to pull from registry '{}'".format(image, login_server))
+    else:
+        raise CLIError("'Blob identified by image '{}' cannot be pulled from registry '{}'".format(image, login_server)) 
 
 # General command
 def acr_check_health(cmd,  # pylint: disable useless-return
                      vnet=None,
                      ignore_errors=False,
                      yes=False,
-                     registry_name=None):
+                     registry_name=None,
+                     repository_name=None,
+                     image=None):
     from azure.cli.core.util import in_cloud_console
+
```

Comment:
You're doing get_registry_by_name 3 times (once in  validate_repository_and_image,  once _check_blob_exists, once in_check_registry_health).

You're doing  get_access_credentials twice (once in  validate_repository_and_image,  once _check_blob_exists). 

Can we just do this once in acr_check_health and pass the info through to the sub functions? This will both cut down on unnecessary network calls and reused code.

Reply:
Refactored this so that get_registry_by_name and get_access_credentials are in acr_check_health and eliminating duplicate calls, thanks for the fix!